donut.c

```diff
@@ -0,0 +1,49 @@
+#include <stdio.h>
+#include <math.h>
+#include <stdint.h>
+#include <string.h>
+#include <unistd.h>
+
+int main() {
+    float A = 0, B = 0;
+    float i, j;
+    int k;
```

Comment:
Consider using `const` for the size of the arrays `b` and `z`. This makes the code more readable and can help the compiler optimize better.